pennylane/transforms/dynamic_one_shot.py

```diff
@@ -209,10 +209,13 @@ def init_auxiliary_tape(circuit: qml.tape.QuantumScript):
             else:
                 new_measurements.append(m)
     for op in circuit.operations:
-        if "MidCircuitMeasure" in str(type(op)):  # pragma: no cover
-            new_measurements.append(qml.sample(op.out_classical_tracers[0]))
-        elif isinstance(op, MidMeasureMP):
-            new_measurements.append(qml.sample(MeasurementValue([op], lambda res: res)))
+        if isinstance(op, MidMeasureMP):
+            mv = (
+                op.mcm_tracer
```

Comment:
Did we not want to use `mv` to the store the tracer then?

Reply:
I think that's a mildly larger change since ideally I would also update PennyLane to use that at the same time, but I'd like to push this across right now to get the Catalyst side working.

Reply:
I'm wondering whether `mcm_tracer` isn't better, at least for now. `mv` makes me think of a `MeasurementValue` object which isn't concrete but tracks all possible values.

Reply:
In my mind `MeasurementValue` is just PennyLane's version of a tracer for measurement values (since it serves the exact same purpose, recording computation) 😅

Reply:
Turns out there are basically no places where we would actually benefit from using `MidMeasureMP.mv` outside of `dynamic_one_shot` and catalyst, so I updated this. Although, I'm now wondering if it makes sense to do this since the places where this is used are so limited and it breaks what `MeasurementProcess.mv` has been used for everywhere else.

Reply:
It does seem to simplify things nicely though :D